### ISSUE TITLE
Freeze when trying to do an expedition in Sky Island (with gdb trace)

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#73224]
### Describe the bug

See title.

### Attach save file

[Sky Island-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/15085356/Sky.Island-trimmed.tar.gz)

I can not reproduce with fresh saves and I actually had to abandon this one because of the freeze.

### Steps to reproduce

1. Load the save.
2. Go to the warp thing.
3. Do a quick expedition.
4. Freeze.

[gdb.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/15085365/gdb.txt)

### Expected behavior

No freeze. This is potentially not a "true" freeze. It probably happens because something *extremely* performance-intensive happens and it would take some minutes to resolve. However, since not even the window is responsive at this point, I decided not to test that.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: cdda-experimental-2024-04-23-1301 9251ce4 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Sky Island [skyisland]
]

**Comment by PatrikLundell** [comment]
I tried it, but it seems to be something resource intensive rather than a hang, as it took maybe a minute to get a "You've arrived! ..." message. Patience...

**Comment by NetSysFire** [comment]
I am patient but it should not take that long nor should the game freeze until it is done with no feedback or progress bar.

**Comment by PatrikLundell** [comment]
I agree it's slower than it ought to be, but it's not hanging at least. Bad performance ought to be addressed as well, but it's a different kind of problem from an infinite loop.
Interestingly enough, the popup appearing when it's done with its walkabout says that the next step should take a bit of time, but didn't seem to do that...
